otelcollector/prometheuscollector/prometheus-ccp-config-merger.go

```diff
@@ -0,0 +1,289 @@
+package main
+
+import (
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"reflect"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	mergedDefaultConfigPath               = "/opt/defaultsMergedConfig.yml"
+	replicasetControllerType              = "replicaset"
+	defaultPromConfigPathPrefix           = "/opt/microsoft/otelcollector/default-prom-configs/"
+	regexHashFile                         = "/opt/microsoft/configmapparser/config_def_targets_metrics_keep_list_hash"
+	controlplaneApiserverDefaultFile      = defaultPromConfigPathPrefix + "controlplane_apiserver.yml"
+	controlplaneKubeSchedulerDefaultFile  = defaultPromConfigPathPrefix + "controlplane_kube_scheduler.yml"
```

Comment:
I get an error when I don't provide a configmap:
<img width="1379" alt="image" src="https://github.com/Azure/prometheus-collector/assets/79080068/092bdc9f-1279-4220-b7d6-f7c5f2ba5710">

Note: I created a [branch](https://github.com/Azure/prometheus-collector/pull/765/files) based on this version but with the default configurations reverted.

Reply:
When I provide a configmap I see this error (I updated the one I was using with the one in the repo just in case)
<img width="1282" alt="image" src="https://github.com/Azure/prometheus-collector/assets/79080068/91353957-8cb0-4810-940d-f486fcd717f8">

Reply:
synced with Nina and we fixed the configmap issue (indention problem).